xapian-applications/omega/omega.cc

```diff
@@ -448,17 +465,34 @@ try {
     // Percentage relevance cut-off
     val = cgi_params.find("THRESHOLD");
     if (val != cgi_params.end()) {
-	threshold = atoi(val->second.c_str());
-	if (threshold < 0) threshold = 0;
-	if (threshold > 100) threshold = 100;
+	unsigned int temp;
+	if (val->second[0] == '-') {
+	    if (!parse_unsigned(val->second.c_str() + 1, temp)) {
+		throw "THRESHOLD parameter must be an integer";
+	    }
+	    else {
```

Comment:
There's no need for an `else` block here since the block above transfers control away via `throw` (this is covered in the Xapian coding style docs).